numpy/polynomial/chebyshev.py

```diff
@@ -856,10 +856,8 @@ def chebpow(c, pow, maxpower=16):
         raise ValueError("Power must be a non-negative integer.")
     elif maxpower is not None and power > maxpower:
         raise ValueError("Power is too large")
-    elif power == 0:
-        return np.array([1], dtype=c.dtype)
     else:
-        return reduce(chebmul, [c]*power)
```

Comment:
There's a reason this `power==0` case exists - it makes it more efficient.

Reply:
I suppose it does... but from the test cases I threw at it, it save a grand total of around 200ns at best. If you prefer the `power==0` check because it is more explicit, or can show how it makes a non-negligible the concern would make more sense to me.